File: backend/src/Domain/Settings/Entities/HoursOfOperationDay.cs
using System;

namespace DetailingArsenal.Domain {
    public class HoursOfOperationDay : Entity<HoursOfOperationDay> {
        public Guid HoursOfOperationId { get; set; } = Guid.Empty;
        public int Day { get; set; }
        public int Open {
            get => open;
            set {
                if (value < 0 || value > 24 * 60) {
                    throw new ArgumentOutOfRangeException();
                }

                open = value;
            }
        }
        public int Close {
            get => close;
            set {
                if (value < 0 || value > 24 * 60) {
                    throw new ArgumentOutOfRangeException();
                }

                close = value;
            }
        }
        public bool Enabled { get; set; }

        private int open;
        private int close;
    }
}

File: app/src/modules/settings/api/entities/hours-of-operation-day.ts
import { Entity } from '@/core';

export class HoursOfOperationDay extends Entity {
    constructor(public day: number, public open: number, public close: number, public enabled: boolean = true) {
        super();
    }

    containsTime(time: number) {
        return time >= this.open && time < this.close;
    }
}
